api/PayrollProcessor.Core.Domain/Intrastructure/Operations/Commands/ICommandDispatcher.cs

```diff
@@ -0,0 +1,12 @@
+using System.Threading;
+using LanguageExt;
+
+namespace PayrollProcessor.Core.Domain.Intrastructure.Operations.Commands
+{
+    public interface ICommandDispatcher
+    {
+        TryOptionAsync<Unit> Dispatch(ICommand command, CancellationToken token = default);
+
+        TryOptionAsync<TResponse> Dispatch<TResponse>(ICommand<TResponse> command, CancellationToken token = default);
```

Comment:
I fixed the issue with previous attempts at these dispatchers/handlers not passing through the `CancellationToken`.

This time all of them pass it through (though the Ardalis.Endpoints doesn't accept it in the method signature, so we'll need to PR into that library to get cancellation added from end-to-end).